code/game/objects/structures/girders.dm

```diff
@@ -38,18 +38,25 @@
 		if(GIRDER_TRAM)
 			. += span_notice("[src] is designed for tram usage. Deconstructed with a screwdriver!")
 
-/obj/structure/girder/attackby(obj/item/W, mob/user, params)
+/obj/structure/girder/attackby(obj/item/attacking_item, mob/user, params)
 	var/platingmodifier = 1
 	if(HAS_TRAIT(user, TRAIT_QUICK_BUILD))
 		platingmodifier = 0.7
 		if(next_beep <= world.time)
 			next_beep = world.time + 10
 			playsound(src, 'sound/machines/clockcult/integration_cog_install.ogg', 50, TRUE)
 	add_fingerprint(user)
-
-	if(istype(W, /obj/item/gun/energy/plasmacutter))
+	var/static/list/acceptable_cutters = list(
+		/obj/item/gun/energy/plasmacutter,
+		/obj/item/melee/energy/sword,
+		/obj/item/dualsaber,
+		/obj/item/pen/edagger,
+	)
+	if(is_type_in_list(attacking_item, acceptable_cutters))
```

Comment:
Why do we have any of this at all instead of making it a generic welding action? Plasmacutters work as welders too right?

Reply:
Because welders dont do this

Reply:
and lighters shouldnt